ckan/lib/plugins.py

```diff
@@ -663,3 +663,41 @@ def get_user_dataset_labels(self, user_obj: model.User) -> list[str]:
             labels.extend('collaborator-%s' % d['package_id'] for d in datasets)
 
         return labels
+
+    def get_activity_labels(self, activity_obj: Activity) -> list[str]:
+        dataset = model.Package.get(activity_obj.object_id)
+        labels = []
+        if not dataset:
+            return labels
+
+        if dataset.state == u'active' and not dataset.private and ('package' in activity_obj.data and not activity_obj.data['package']['private']):
+            labels.append(u'public')
+
+        if ckan.authz.check_config_permission('allow_dataset_collaborators'):
+            labels.append(u'collaborator-%s' % dataset.id)
+
+        if dataset.owner_org:
+            labels.append(u'member-%s' % dataset.owner_org)
+        else:
+            labels.append(u'creator-%s' % dataset.creator_user_id)
+
+        return labels
+
+    def get_user_activity_labels(self, user_obj: model.User) -> list[str]:
+        labels = [u'public']
+        if not user_obj or user_obj.is_anonymous:
+            return labels
+
+        labels.append(u'creator-%s' % user_obj.id)
+
+        orgs = logic.get_action(u'organization_list_for_user')(
+            {u'user': user_obj.id}, {u'permission': u'read'})
+        labels.extend(u'member-%s' % o[u'id'] for o in orgs)
+
+        if ckan.authz.check_config_permission('allow_dataset_collaborators'):
+            datasets = logic.get_action('package_collaborator_list_for_user')(
+                {'ignore_auth': True}, {'id': user_obj.id})
+
+            labels.extend('collaborator-%s' % d['package_id'] for d in datasets)
+
+        return labels
```

Comment:
Importing `ckanext.activity.model` .. from core doesn't sit well with me.  @smotornyuk WDYT? Since the only labels generated and tested here are a generated with a copy of the same code used in the dataset labels, would we be better off without the new methods and just assuming that the only permission checks required will be based on dataset-activities, or should we create a new interface for activity permissions in the activity extension? We could start with dataset-only and think about adding a new interface when there is a specific need for one.

Reply:
Just confirmed with the team: It would be better to remove these methods and call the existing dataset ones for dataset-related activities from the activity extension instead.